test/integration/confirmations/signatures/permit.test.tsx

```diff
@@ -0,0 +1,239 @@
+  return {
+    ...mockMetaMaskState,
+    preferences: {
+      ...mockMetaMaskState.preferences,
+      redesignedConfirmationsEnabled: true,
+    },
+    unapprovedTypedMessages: {
+      [pendingPermitId]: {
+        id: pendingPermitId,
+        status: 'unapproved',
+        time: pendingPermitTime,
+        type: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA,
+        securityProviderResponse: null,
+        msgParams: {
+          from: accountAddress,
+          version: 'v4',
+          data: `{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Permit":[{"name":"owner","type":"address"},{"name":"spender","type":"address"},{"name":"value","type":"uint256"},{"name":"nonce","type":"uint256"},{"name":"deadline","type":"uint256"}]},"primaryType":"Permit","domain":{"name":"MyToken","version":"1","verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC","chainId":1},"message":{"owner":"${accountAddress}","spender":"0x5B38Da6a701c568545dCfcB03FcB875f56beddC4","value":3000,"nonce":0,"deadline":50000000000}}`,
+          origin: 'https://metamask.github.io',
+          siwe: { isSIWEMessage: false, parsedMessage: null },
+          signatureMethod: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA_V4,
+        },
+      },
+    },
+    unapprovedTypedMessagesCount: 1,
+    pendingApprovals: {
+      [pendingPermitId]: {
+        id: pendingPermitId,
+        origin: 'origin',
+        time: pendingPermitTime,
+        type: ApprovalType.EthSignTypedData,
+        requestData: {
+          from: accountAddress,
+          version: 'v4',
+          data: `{"types":{"EIP712Domain":[{"name":"name","type":"string"},{"name":"version","type":"string"},{"name":"chainId","type":"uint256"},{"name":"verifyingContract","type":"address"}],"Permit":[{"name":"owner","type":"address"},{"name":"spender","type":"address"},{"name":"value","type":"uint256"},{"name":"nonce","type":"uint256"},{"name":"deadline","type":"uint256"}]},"primaryType":"Permit","domain":{"name":"MyToken","version":"1","verifyingContract":"0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC","chainId":1},"message":{"owner":"${accountAddress}","spender":"0x5B38Da6a701c568545dCfcB03FcB875f56beddC4","value":3000,"nonce":0,"deadline":50000000000}}`,
+          origin: 'https://metamask.github.io',
+          signatureMethod: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA_V4,
+          metamaskId: pendingPermitId,
+        },
+        requestState: null,
+        expectsResult: false,
+      },
+    },
+    pendingApprovalCount: 1,
+  };
+};
+
+describe('Permit Confirmation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    mockedBackgroundConnection.submitRequestToBackground.mockImplementation(
+      createMockImplementation('getTokenStandardAndDetails', { decimal: '10' }),
+    );
+  });
+
+  it('displays the header account modal with correct data', async () => {
```

Comment:
Do you think it is necessary to assert confirmation header and modal behaviour on every confirmation type? 🤔 
I was thinking that we could assert the header info, but the modal behaviour we would only assert on one of the tests... though not sure about it. These integration tests are also cheap to run, so we might go for full coverage. What do you think?

Reply:
yeah the tests are super cheap and fast so no issues in checking them everywhere. There are small differences for e.g. in the metrics event the signature_type depends on the type of confirmation etc. these things would warrant keeping the test here instead of separating them out into a separate file.
The only downside I can think of is if the modal behavior changes we might have to change it in all tests.
I would say for now let's keep it this way as we do it in the e2e as well and then if we refactor the header component at any point we can see if we want to refactor and separate the tests? WDYT?

Reply:
Sounds good 👍